### PR TITLE
Fixed #26795 -- Factor out "get_changes" method in test_autodetector

### PR DESCRIPTION
**Pull request by akki**
Ticket [#26795](https://code.djangoproject.com/ticket/26795)

I have added `MigrationQuestioner` argument to `get_changes` in my second commit to incorporate other use cases as well.

### CODE REVIEW

File: tests/migrations/test_autodetector.py [hunk 1]
```diff
@@ -550,6 +550,16 @@ def make_project_state(self, model_states):
             project_state.add_model(model_state.clone())
         return project_state
 
+    def get_changes(self, before_states, after_states, migration_questioner=None):
```
Comment (timgraham) [inline tests/migrations/test_autodetector.py]: migration_questioner -> questioner

File: tests/migrations/test_autodetector.py [hunk 2]
```diff
@@ -550,6 +550,16 @@ def make_project_state(self, model_states):
...
+        args = []
```
Comment (timgraham) [inline tests/migrations/test_autodetector.py]:
```
args = [
    self.make_project_state(before_states),
    self.make_project_state(after_states),
]
```

File: tests/migrations/test_autodetector.py [hunk 3]
```diff
@@ -762,13 +738,11 @@ def content_file_name(key, **kwargs):
             (_content_file_name, ('other-file',), {}),
             (value.func, value.args, value.keywords)
         )
-
```
Comment (timgraham) [inline tests/migrations/test_autodetector.py]: the blank line here seems to be separating logical parts of the test so I wouldn't remove it

File: tests/migrations/test_autodetector.py [hunk 4]
```diff
@@ -862,11 +822,11 @@ def test_rename_m2m_through_model(self):
...
         """
-        # Make state
-        before = self.make_project_state([self.author_with_m2m_through, self.publisher, self.contract])
-        after = self.make_project_state([self.author_with_renamed_m2m_through, self.publisher, self.contract_renamed])
-        autodetector = MigrationAutodetector(before, after, MigrationQuestioner({'ask_rename_model': True}))
-        changes = autodetector._detect_changes()
+        changes = self.get_changes(
+            [self.author_with_m2m_through, self.publisher, self.contract],
+            [self.author_with_renamed_m2m_through, self.publisher, self.contract_renamed],
+            MigrationQuestioner({"ask_rename_model": True})
```
Comment (timgraham) [inline tests/migrations/test_autodetector.py]: Should be changing double quotes to single, if at all. I wouldn't really worry about it though.

File: tests/migrations/test_autodetector.py [hunk 5]
```diff
@@ -1742,13 +1557,10 @@ def test_many_to_many_removed_before_through_model(self):
         Removing a ManyToManyField and the "through" model in the same change
         must remove the field before the model to maintain consistency.
         """
-        before = self.make_project_state([
+        changes = self.get_changes([
             self.book_with_multiple_authors_through_attribution, self.author_name, self.attribution
```
Comment (timgraham) [inline tests/migrations/test_autodetector.py]: use the same style as before:
```
self.get_changes([
    [...],
    [...],
)
```

**Comment by timgraham** [comment]
A nice change. You may squash commits when updating.

**Comment by akki** [comment]
Done.

**Pull request merged**